src/treeViews/currentBranch.ts

```diff
@@ -112,7 +112,15 @@ export class CurrentBranchTreeProvider
 
     const resp = result.data;
     const runs = resp.workflow_runs;
+    // We are removing newlines from workflow names for presentation purposes
+    const runsWithNamesWithNoNewlines = runs.map(run => {
```

Comment:
Minor: We're using `map` here to create new a array of runs, but we're still modifying the underlying run objects.

Since these runs are from an API response we called in this function, I think it's fine to modify the `run` variable and just do this in a loop:

```ts
    // We are removing newlines from workflow names for presentation purposes
    for (const run of runs) {
      run.name = run.name?.replace(/(\r\n|\n|\r)/gm, " ");
    }

    return this.runNodes(gitHubRepoContext, runs, true);
```

Reply:
If you did want to copy the runs (which I don't think you need to do here), you could do a shallow copy with a `...` spread

```ts
    const runsWithNamesWithNoNewlines = runs.map(run => {
      return {
        ...run,
        name: run.name?.replace(/(\r\n|\n|\r)/gm, " ")
      };
    });
```